src/main/kotlin/org/rust/toml/CargoTomlDependenciesCompletionProvider.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml
+
+import com.intellij.codeInsight.completion.*
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.ext.ancestorOrSelf
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.isAncestorOf
+import org.toml.lang.psi.*
+
+abstract class AbstractTomlKeyValueCompletionProvider : CompletionProvider<CompletionParameters>() {
```

Comment:
Partially resolved: I don't think that the name should contains "Dependency" because this class can be used for any TomlKeyValue completion. I removed any mentions of "Dependency" word from this class